### PR TITLE
Add exit code metrics

### PR DESCRIPTION
**Pull request by fabriziodemaria**

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 1]
```diff
@@ -106,7 +112,8 @@
   private final ConcurrentMap<String, Meter> storageOperationMeters;
   private final ConcurrentMap<String, Histogram> dockerOperationHistograms;
   private final ConcurrentMap<String, Meter> dockerOperationMeters;
-  private final ConcurrentHashMap<WorkflowId, Gauge> activeStatesPerWorkflowGauges;
+  private final ConcurrentMap<WorkflowId, Gauge> activeStatesPerWorkflowGauges;
+  private final ConcurrentMap<Tuple2<WorkflowId, Integer>, Meter> exitCodePerWorkflowMeters;
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: As we talked, I actually prefer the nested map over tuple.

File: styx-scheduler-service/src/test/java/com/spotify/styx/state/handlers/MonitoringHandlerTest.java [hunk 2]
```diff
@@ -71,6 +73,21 @@ public void shouldUpdateOnDefaultCase() throws Exception {
   }
 
   @Test
+  public void shouldMarkExitCode() throws Exception {
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/state/handlers/MonitoringHandlerTest.java]: Another test case to cover the other branch? If there is no exit code.

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 3]
```diff
@@ -53,6 +55,10 @@
       .tagged("what", "workflow-count")
       .tagged("unit", "workflow");
 
+  private static final MetricId EXIT_CODE_RATE = BASE
+      .tagged("what", "exit-code-rate")
+      .tagged("unit", "code");
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: `code` as unit feels strange, we can leave it as default which will be `n`.

File: styx-scheduler-service/src/test/java/com/spotify/styx/state/handlers/MonitoringHandlerTest.java (hunk 2 above)
Comment (fabriziodemaria) [inline styx-scheduler-service/src/test/java/com/spotify/styx/state/handlers/MonitoringHandlerTest.java]: Added

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java (hunk 1 above)
Comment (fabriziodemaria) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: We talked about it IRL and I think we agreed on keeping the tuple keys? @honnix

Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: yup

**Comment by honnix** [comment]
👍

**Pull request merged**